packages/react/src/components/Geo/AmplifyMap/index.tsx

```diff
@@ -0,0 +1,75 @@
+import type { ICredentials } from '@aws-amplify/core';
+import { Amplify, Auth } from 'aws-amplify';
+import maplibregl from 'maplibre-gl';
+import { AmplifyMapLibreRequest } from 'maplibre-gl-js-amplify';
+import React, { useEffect, useState } from 'react';
+import ReactMapGL from 'react-map-gl';
+import type { MapProps, TransformRequestFunction } from 'react-map-gl';
+
+/**
+ * The `AmplifyMap` component uses [react-map-gl](https://visgl.github.io/react-map-gl/) and
+ * [maplibre-gl-js](https://visgl.github.io/react-map-gl/) to provide an interactive map using
+ * [Amplify Geo APIs](https://docs.amplify.aws/lib/geo/getting-started/q/platform/js/) powered by
+ * [Amazon Location Service](https://aws.amazon.com/location/). Since `AmplifyMap` is a wrapper of the
+ * [react-map-gl default Map](https://visgl.github.io/react-map-gl/docs/api-reference/map), it accepts the same
+ * properties.
+ *
+ * @example
+ * // Basic usage of AmplifyMap:
+ * function App() {
+ *   return <AmplifyMap />
+ * }
+ */
+export const AmplifyMap = ({ style, ...props }: MapProps) => {
+  const [transformRequest, setTransformRequest] = useState<
+    TransformRequestFunction | undefined
+  >();
+  const [credentials, setCredentials] = useState<ICredentials | undefined>();
+
+  const amplifyConfig = Amplify.configure() as any;
+  const mapId = amplifyConfig.geo?.amazon_location_service.maps.default;
+  const region = amplifyConfig.geo?.amazon_location_service.region;
+
+  const styleProps = {
+    height: '100vh',
+    position: 'relative',
+    width: '100vw',
+    ...style,
+  } as React.CSSProperties;
+
+  useEffect(() => {
+    const getCredentials = async () => {
+      setCredentials(await Auth.currentUserCredentials());
+    };
+
+    getCredentials();
+  }, []);
```

Comment:
A note about credentials and amplify config:

I originally had a Geo xstate machine built out to manage map and geocoder state which also handled fetching credentials and Amplify configurations. However, after updating to `react-map-gl` v7, I was able to greatly reduce the amount of state we would need to manage for our components. Thus, I removed the Geo xstate machine implementation since it doesn't currently solve any problems for our Geo components, and fetched credentials and config within the component itself as I need them.

Until the use of xstate provides a benefit, I'm of the opinion that we should not implement it and favor simplified react components.